src/interactivity/win32/ut_interactivity_win32/UiaTextRangeTests.cpp

```diff
@@ -1136,57 +1136,82 @@ class UiaTextRangeTests
+        VERIFY_ARE_EQUAL(0, moveAmt);
 
         // Verify that moving backwards still works properly
+
+        // write "temp" at (2,2)
         const COORD writeTarget{ 2, 2 };
         _pTextBuffer->Write({ L"temp" }, writeTarget);
-        for (int unit = TextUnit::TextUnit_Character; unit != TextUnit::TextUnit_Document; ++unit)
-        {
-            COORD expectedEnd;
-            switch (static_cast<TextUnit>(unit))
-            {
-            case TextUnit::TextUnit_Character:
-                expectedEnd = endInclusive;
-                break;
-            case TextUnit::TextUnit_Word:
-                expectedEnd = writeTarget;
-                break;
-            case TextUnit::TextUnit_Line:
-                expectedEnd = endExclusive;
-                break;
-            case TextUnit::TextUnit_Document:
-                expectedEnd = bufferSize.Origin();
-                break;
-            default:
-                continue;
-            }
 
-            Log::Comment(NoThrowString().Format(L"Backwards by %s", toString(static_cast<TextUnit>(unit))));
+        //if (textUnit == TextUnit::TextUnit_Character && !degenerate)
+        //    DebugBreak();
 
-            // Create a degenerate UTR at EndExclusive
-            THROW_IF_FAILED(Microsoft::WRL::MakeAndInitialize<UiaTextRange>(&utr, _pUiaData, &_dummyProvider, endInclusive, endExclusive));
-            THROW_IF_FAILED(utr->Move(static_cast<TextUnit>(unit), -1, &moveAmt));
+        Log::Comment(NoThrowString().Format(L"Backwards by %s", toString(textUnit)));
+        THROW_IF_FAILED(utr->Move(textUnit, -1, &moveAmt));
 
-            VERIFY_ARE_EQUAL(expectedEnd, utr->_end);
+        // NOTE: If the range is degenerate, _start == _end before AND after the move.
+        if (textUnit <= TextUnit::TextUnit_Character)
+        {
+            // Special case: _end will always be endInclusive, because...
+            // -  degenerate --> it moves with _start to stay degenerate
+            // - !degenerate --> it excludes the last char, to select the second to last char
+            VERIFY_ARE_EQUAL(degenerate ? endInclusive : secondToLastCharacterPos, utr->_start);
+            VERIFY_ARE_EQUAL(endInclusive, utr->_end);
+            VERIFY_ARE_EQUAL(-1, moveAmt);
+        }
+        else if (textUnit <= TextUnit::TextUnit_Word)
+        {
+            VERIFY_ARE_EQUAL(origin, utr->_start);
+            VERIFY_ARE_EQUAL(degenerate ? origin : writeTarget, utr->_end);
+            VERIFY_ARE_EQUAL(-1, moveAmt);
+        }
+        else if (textUnit <= TextUnit::TextUnit_Line)
+        {
+            VERIFY_ARE_EQUAL(lastLineStart, utr->_start);
+            VERIFY_ARE_EQUAL(degenerate ? lastLineStart : endExclusive, utr->_end);
+            VERIFY_ARE_EQUAL(-1, moveAmt);
+        }
+        else // textUnit <= TextUnit::TextUnit_Document:
+        {
+            VERIFY_ARE_EQUAL(origin, utr->_start);
+            VERIFY_ARE_EQUAL(degenerate ? origin : endExclusive, utr->_end);

```

Comment:
i know it might be a chore, but having @codeofdusk review this test case for sanity would be helpful

Reply:
(a chore for him, not a chore for you. since this is a bunch of dense WEX C++)